ctapipe/image/tests/test_extractor.py

```diff
@@ -617,3 +635,46 @@ def test_global_peak_window_sum_with_pixel_fraction(subarray):
 
     expected = np.average([29, 30, 31], weights=[5, 10, 3])
     assert np.allclose(dl1.peak_time[bright_pixels], expected / sample_rate)
+
+
+def test_flashcam_extractor(toymodel_1_MST_FC, prod5_gamma_simtel_path):
+    # Test on toy model
+    (
+        waveforms,
+        subarray,
+        tel_id,
+        selected_gain_channel,
+        true_charge,
+        true_time,
+    ) = toymodel_1_MST_FC
+    extractor = FlashCamExtractor(subarray=subarray)
+    broken_pixels = np.zeros(waveforms.shape[0], dtype=bool)
+    dl1 = extractor(waveforms, tel_id, selected_gain_channel, broken_pixels)
+    assert dl1.is_valid == True
+
+    assert_allclose(dl1.image, true_charge, rtol=0.15)
+    assert_allclose(dl1.peak_time, true_time, atol=1.0)
+
+    # Test on prod5 simulations
+    with EventSource(prod5_gamma_simtel_path) as source:
+        subarray = source.subarray
+        extractor = FlashCamExtractor(subarray)
+        for event in source:
+            for tel_id in subarray.get_tel_ids_for_type("MST_MST_FlashCam"):
+                true_charge = event.simulation.tel[tel_id].true_image
+                if true_charge is None:
+                    continue  # telescope did not trigger
```

Comment:
Or the input file does not contain true images! If you want to check something, you should do expicitly the check for what you are checking, i.e. here it would be:

```
if tel_id in event.trigger.tels_with_trigger
```

Or maybe just loop over the `tels_with_trigger`,  using a filter function to only yield flashcam ids:

```
def is_flashcam(tel_id):
    return subarray.tel[1].camera.name == "FlashCam" 

for tel_id in filter(is_flashcam, event.trigger.tels_with_trigger):
    
```